### PR TITLE
[dhcp_relay] Add stress test for restarting dhcp_relay

### PR DESCRIPTION
**Pull request by yaqiangz**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [x] 202311
- [x] 202405

#### What is the motivation for this PR?
Address test gap for this change: https://github.com/sonic-net/sonic-buildimage/pull/20021
Prevously, dhcrelay would hit the issue that it wouldn't relay any packets if there are packets come when dhcrelay startup. This issue has been fixed from image side by https://github.com/sonic-net/sonic-buildimage/pull/20021. This PR is to add test for it.

#### How did you do it?
Add stress test with dhcp_relay restart:
1) Keep sending DHCP packets
2) Restart dhcp_relay
3) Check socket buffer
4) Run general dhcp relay test.

#### How did you verify/test it?
Run test on m0/t0/dualtor topos, all passed

### CODE REVIEW

**Review by Blueve — APPROVED** [review]

File: tests/dhcp_relay/test_dhcp_relay_stress.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+from tests.common.fixtures.ptfhost_utils import copy_ptftests_directory   # noqa F401
+from tests.common.dualtor.mux_simulator_control import toggle_all_simulator_ports_to_rand_selected_tor_m    # noqa F401
+from tests.dhcp_relay.dhcp_relay_utils import restart_dhcp_service
+from tests.common.helpers.assertions import pytest_assert, pytest_require
+from tests.common.utilities import wait_until
+from tests.ptf_runner import ptf_runner
+
+pytestmark = [
+    pytest.mark.topology('t0', 'm0'),
+    pytest.mark.device_type('vs')
```
Comment (yutongzhang-microsoft) [inline tests/dhcp_relay/test_dhcp_relay_stress.py:13]: Please add this new script into PR test.

Comment (yutongzhang-microsoft) [reply tests/dhcp_relay/test_dhcp_relay_stress.py:13]: You can add this script name into https://github.com/sonic-net/sonic-mgmt/blob/master/.azure-pipelines/pr_test_scripts.yaml, per to its support topology.

Comment (yaqiangz) [reply tests/dhcp_relay/test_dhcp_relay_stress.py:13]: Will add it later

**Review by wangxin — APPROVED** [review]

**Comment by yaqiangz** [comment]
Hi @yxieca @bingwang-ms could you please help to approve the backport request?

**Pull request merged**